extensions-core/lookups-cached-global/src/main/java/org/apache/druid/server/lookup/namespace/cache/CacheScheduler.java

```diff
@@ -387,10 +379,11 @@ public String toString()
     final CacheHandler cacheHandler;
     final String version;
 
-    private VersionedCache(String entryId, String version)
+    private VersionedCache(String entryId, String version, CacheHandler cache)
     {
       this.entryId = entryId;
-      this.cacheHandler = cacheManager.createCache();
+      this.cacheHandler = cache;
+      cacheManager.attachCache(cache);
```

Comment:
This is doing work in the constructor.  That's a pattern that often creates sadness.  Given that `cacheManager` is a field on the parent object here, can we move the attachCache into the `swapCacheState` method maybe?